src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchSource.java

```diff
@@ -164,7 +165,10 @@ public static List<SalesforceSplit> getSplits(
       bulkConnection.addHeader(SalesforceSourceConstants.HEADER_ENABLE_PK_CHUNK,
           String.join(";", chunkHeaderValues));
     }
-    List<SalesforceSplit> querySplits = SalesforceSplitUtil.getQuerySplits(query, bulkConnection,
+    BulkConnectionRetryWrapper bulkConnectionRetryWrapper = new BulkConnectionRetryWrapper(bulkConnection,
```

Comment:
Rather than passing the bulkConnection here, write a method in SalesforceSplitUtil that returns RetryWrapper